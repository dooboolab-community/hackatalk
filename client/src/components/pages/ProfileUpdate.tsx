import {Alert, Image, TouchableOpacity, View} from 'react-native';
import {Button, EditText, useTheme} from 'dooboo-ui';
import {IC_CAMERA, IC_PROFILE} from '../../utils/Icons';
import React, {FC, useCallback, useEffect, useRef, useState} from 'react';
import {
  UploadSingleUploadMutation,
  UploadSingleUploadMutationResponse,
} from '../../__generated__/UploadSingleUploadMutation.graphql';
import {fetchQuery, useMutation, useRelayEnvironment} from 'react-relay';
import {
  launchCameraAsync,
  launchImageLibraryAsync,
} from '../../utils/ImagePicker';
import {meQuery, profileUpdate} from '../../relay/queries/User';

import {ImagePickerResult} from 'expo-image-picker';
import {ReactNativeFile} from 'apollo-upload-client';
import type {UserMeQuery} from '../../__generated__/UserMeQuery.graphql';
import type {UserUpdateProfileMutation} from '../../__generated__/UserUpdateProfileMutation.graphql';
import {getString} from '../../../STRINGS';
import {resizePhotoToMaxDimensionsAndCompressAsPNG} from '../../utils/image';
import {showAlertForError} from '../../utils/common';
import {singleUpload} from '../../relay/queries/Upload';
import styled from '@emotion/native';
// import {uploadImageAsync} from '../../apis/upload';
import {useActionSheet} from '@expo/react-native-action-sheet';
import {useAuthContext} from '../../providers/AuthProvider';

const BUTTON_INDEX_LAUNCH_CAMERA = 0;
const BUTTON_INDEX_LAUNCH_IMAGE_LIBRARY = 1;
const BUTTON_INDEX_CANCEL = 2;

const DEFAULT = {
  PROFILEIMAGE_WIDTH: 1280,
  PROFILEIMAGE_HEIGHT: 1280,
};

const Container = styled.View`
  flex: 1;
  background-color: ${({theme}) => theme.background};
  flex-direction: column;
  align-items: center;
`;

const StyledScrollView = styled.ScrollView`
  align-self: stretch;
`;

const Wrapper = styled.View`
  margin-top: 48px;
  width: 78%;
  flex-direction: column;
  align-items: center;
`;

const StyledButtonWrapper = styled.View`
  flex: 1;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
  align-self: stretch;
  height: 60px;
  margin-top: 36px;
  margin-bottom: 48px;
`;

const ProfileImage = styled.Image`
  width: 90px;
  height: 90px;
  border-radius: 45px;
`;

const Screen: FC = () => {
  const {theme} = useTheme();
  const [name, setName] = useState('');
  const [nickname, setNickname] = useState('');
  const [statusMessage, setstatusMessage] = useState('');
  const {showActionSheetWithOptions} = useActionSheet();
  const [profilePath, setProfilePath] = useState('');
  const environment = useRelayEnvironment();
  const envrionmentProps = useRef(environment);

  const {
    state: {user},
  } = useAuthContext();

  const [
    commitUpload,
    isUploadInFlight,
  ] = useMutation<UploadSingleUploadMutation>(singleUpload);

  const [
    commitProfileUpdate,
    isUpdating,
  ] = useMutation<UserUpdateProfileMutation>(profileUpdate);

  const [isUploading, setIsUploading] = useState(false);

  useEffect(() => {
    fetchQuery<UserMeQuery>(envrionmentProps.current, meQuery, {}).subscribe({
      next: (data) => {
        if (data.me) {
          const {
            name: myName,
            nickname: nickName,
            statusMessage: statusMsg,
            photoURL,
            thumbURL,
          } = data.me;

          setName(myName ?? '');
          setNickname(nickName ?? '');
          setstatusMessage(statusMsg ?? '');
          setProfilePath((thumbURL || photoURL) ?? '');
        }
      },
    });
  }, []);

  const changeText = (type: string, text: string): void => {
    switch (type) {
      case 'NICKNAME':
        setNickname(text);
        break;
      case 'NAME':
        setName(text);
        break;
      case 'STATUS_MSG':
      default:
        setstatusMessage(text);
        break;
    }
  };

  const processImageUpload = useCallback(
    async (image: ImagePickerResult | null): Promise<void> => {
      if (image && !image.cancelled) {
        setIsUploading(true);

        try {
          const resizedImage = await resizePhotoToMaxDimensionsAndCompressAsPNG(
            {
              uri: image.uri,
              width: DEFAULT.PROFILEIMAGE_WIDTH,
              height: DEFAULT.PROFILEIMAGE_HEIGHT,
            },
          );

          const fileName = resizedImage.uri.split('/').pop() || '';
          const fileTypeMatch = /\.(\w+)$/.exec(fileName);

          const fileType = fileTypeMatch
            ? `image/${fileTypeMatch[1]}`
            : 'image';

          const file = new ReactNativeFile({
            uri: resizedImage.uri,
            name: `${user?.id || fileName}.${fileType}`,
            type: fileType,
          });

          commitUpload({
            variables: {
              file: null,
              dir: 'profiles',
            },
            uploadables: {
              file,
            },
            onCompleted: (response: UploadSingleUploadMutationResponse) => {
              const url = response.singleUpload;

              if (url) setProfilePath(url);

              setIsUploading(false);
            },
            onError: (err) => {
              Alert.alert(getString('ERROR'), err.message);
            },
          });
        } catch (err) {
          Alert.alert(getString('ERROR'), getString('FAILED_LOAD_IMAGE'));
        } finally {
          setIsUploading(false);
        }
      }
    },
    [commitUpload, user?.id],
  );

  const pressProfileImage = async (): Promise<void> => {
    const options = [
      getString('TAKE_A_PICTURE'),
      getString('SELECT_FROM_ALBUM'),
      getString('CANCEL'),
    ];

    showActionSheetWithOptions(
      {
        options,
        cancelButtonIndex: BUTTON_INDEX_CANCEL,
      },
      async (buttonIndex: number) => {
        if (buttonIndex === BUTTON_INDEX_LAUNCH_CAMERA) {
          const image = await launchCameraAsync();

          processImageUpload(image);

          return;
        }

        if (buttonIndex === BUTTON_INDEX_LAUNCH_IMAGE_LIBRARY) {
          const image = await launchImageLibraryAsync();

          processImageUpload(image);
        }
      },
    );
  };

  const updateProfile = async (): Promise<void> => {
    const mutationConfig = {
      variables: {
        user: {
          name,
          nickname,
          statusMessage,
          thumbURL: profilePath,
          photoURL: profilePath,
        },
      },
      onError: (error: Error): void => {
        showAlertForError(error);
      },
    };

    commitProfileUpdate(mutationConfig);
  };

  return (
    <Container>
      <StyledScrollView
        contentContainerStyle={{
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <Wrapper>
          <TouchableOpacity
            testID="button-user-icon"
            activeOpacity={0.5}
            onPress={pressProfileImage}>
            {!profilePath ? (
              <View
                style={{
                  width: 90,
                  height: 90,
                }}>
                <Image style={{height: 80, width: 80}} source={IC_PROFILE} />
                <Image
                  style={{
                    position: 'absolute',
                    bottom: 0,
                    right: 0,
                  }}
                  source={IC_CAMERA}
                />
              </View>
            ) : (
              <ProfileImage
                testID="profile-image"
                resizeMode="cover"
                source={{uri: profilePath}}
              />
            )}
          </TouchableOpacity>
          <EditText
            testID="input-nickname"
            style={{marginTop: 32}}
            styles={{
              container: {
                borderColor: theme.text,
              },
              input: {
                color: theme.text,
              },
            }}
            labelText={getString('NICKNAME')}
            placeholder={getString('NICKNAME_HINT')}
            value={nickname}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            onChangeText={(text: string): void => changeText('NICKNAME', text)}
          />
          <EditText
            testID="input-name"
            style={{marginTop: 32}}
            styles={{
              container: {
                borderColor: theme.text,
              },
              input: {
                color: theme.text,
              },
            }}
            labelText={getString('NAME')}
            placeholder={getString('NAME_HINT')}
            value={name}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            onChangeText={(text: string): void => changeText('NAME', text)}
          />
          <EditText
            testID="input-status"
            type="column"
            style={{marginTop: 36}}
            styles={{
              input: {
                marginTop: 12,
                color: theme.text,
              },
              container: {
                borderColor: theme.text,
                borderWidth: 1,
                paddingHorizontal: 8,
                paddingVertical: 12,
              },
            }}
            labelText={getString('STATUS_MSG')}
            placeholder={getString('STATUS_MSG_HINT')}
            value={statusMessage}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            onChangeText={(text: string): void =>
              changeText('STATUS_MSG', text)
            }
            textInputProps={{
              multiline: true,
            }}
          />
          <StyledButtonWrapper>
            <Button
              testID="button-update"
              style={{alignSelf: 'stretch', flex: 1}}
              styles={{
                container: {
                  height: 44,
                  backgroundColor: theme.btnPrimary,
                  borderColor: theme.btnPrimary,
                  borderWidth: 1,
                },
                text: {
                  color: theme.btnPrimaryFont,
                  fontSize: 14,
                  fontWeight: 'bold',
                },
              }}
              loading={isUploading || isUpdating || isUploadInFlight}
              onPress={updateProfile}
              text={getString('UPDATE')}
            />
          </StyledButtonWrapper>
        </Wrapper>
      </StyledScrollView>
    </Container>
  );
};

export default Screen;
