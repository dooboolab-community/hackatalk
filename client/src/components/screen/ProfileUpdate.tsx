import * as Config from '../../../config';

import { Alert, Image, TouchableOpacity, View } from 'react-native';
import { Button, EditText } from 'dooboo-ui';
import { IC_CAMERA, IC_PROFILE } from '../../utils/Icons';
import React, { FC, useEffect, useState } from 'react';
import {
  fetchQuery,
  graphql,
  useRelayEnvironment,
} from 'react-relay/hooks';
import { launchCameraAsync, launchImageLibraryAsync } from '../../utils/ImagePicker';

import AsyncStorage from '@react-native-community/async-storage';
import { EditTextInputType } from 'dooboo-ui/EditText';
import { MainStackNavigationProps } from '../navigation/MainStackNavigator';
import type { ProfileUpdateMeQuery } from '../../__generated__/ProfileUpdateMeQuery.graphql';
import { getString } from '../../../STRINGS';
import { resizeImage } from '../../utils/image';
import styled from 'styled-components/native';
import { useActionSheet } from '@expo/react-native-action-sheet';
import { useThemeContext } from '@dooboo-ui/theme';

const BUTTON_INDEX_LAUNCH_CAMERA = 0;
const BUTTON_INDEX_LAUNCH_IMAGE_LIBLARY = 1;
const BUTTON_INDEX_CANCEL = 2;
const DEFAULT = {
  PROFILEIMAGE_WIDTH: 300,
  PROFILEIMAGE_HEIGHT: 300,
};

const Container = styled.View`
  flex: 1;
  background-color: ${({ theme }): string => theme.background};
  flex-direction: column;
  align-items: center;
`;

const StyledScrollView = styled.ScrollView`
  align-self: stretch;
`;

const Wrapper = styled.View`
  margin-top: 48px;
  width: 78%;
  flex-direction: column;
  align-items: center;
`;

const StyledButtonWrapper = styled.View`
  flex: 1;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
  align-self: stretch;
  height: 60px;
  margin-top: 36px;
  margin-bottom: 48px;
`;

const ProfileImage = styled.Image`
  width: 90px;
  height: 90px;
  border-radius: 45px;
`;

interface Props {
  navigation: MainStackNavigationProps<'ProfileUpdate'>;
}

const meQuery = graphql`
  query ProfileUpdateMeQuery {
    me {
      id
      email
      name
      nickname
      statusMessage
      verified
      profile {
        authType
      }
    }
  }
`;

const Screen: FC<Props> = () => {
  const { theme } = useThemeContext();
  const [name, setName] = useState('');
  const [nickname, setNickname] = useState('');
  const [statusMessage, setstatusMessage] = useState('');
  const { showActionSheetWithOptions } = useActionSheet();
  const [profilePath, setProfilePath] = useState('');
  const environment = useRelayEnvironment();

  useEffect(() => {
    fetchQuery<ProfileUpdateMeQuery>(environment, meQuery, {}).subscribe({
      next: (data) => {
        if (data.me) {
          const { name, nickname, statusMessage } = data.me;
          setName(name ?? '');
          setNickname(nickname ?? '');
          setstatusMessage(statusMessage ?? '');
        }
      },
    });
  }, []);

  const changeText = (type: string, text: string): void => {
    switch (type) {
      case 'NICKNAME':
        setNickname(text);
        break;
      case 'NAME':
        setName(text);
        break;
      case 'STATUS_MSG':
      default:
        setstatusMessage(text);
        break;
    }
  };

  const uploadImage = async (uri: string): Promise<string> => {
    const fileName = uri.split('/').pop() || '';
    const fileTypeMatch = /\.(\w+)$/.exec(fileName);
    const fileType = fileTypeMatch ? `image/${fileTypeMatch[1]}` : 'image';
    const data: FormData = new FormData();
    const token = await AsyncStorage.getItem('token');
    data.append('profile', {
      uri: uri,
      type: fileType,
      name: fileName,
    });

    const fetchInitOption = {
      method: 'POST',
      body: {
        inputFile: data,
        dir: 'profiles',
      },
      headers: new Headers({
        Accept: 'application/json',
        Authorization: `Bearer ${token}`,
      }),
    };

    return fetch(`${Config.ROOT_URL}/upload_single`, fetchInitOption)
      .then((res: Response) => res.url)
      .catch(() => {
        Alert.alert(getString('ERROR'), getString('ERROR_OCCURED'));
        return '';
      });
  };

  const pressProfileImage = async (): Promise<void> => {
    const options = [
      getString('TAKE_A_PICTURE'),
      getString('SELSCT_FROM_ALBUM'),
      getString('CANCEL'),
    ];

    showActionSheetWithOptions(
      {
        options,
        cancelButtonIndex: BUTTON_INDEX_CANCEL,
      },
      async (buttonIndex: number) => {
        if (buttonIndex === BUTTON_INDEX_LAUNCH_CAMERA) {
          const image = await launchCameraAsync();
          if (image && !image.cancelled) {
            const resizedImage = await resizeImage({
              imageUri: image.uri,
              maxWidth: DEFAULT.PROFILEIMAGE_WIDTH,
              maxHeight: DEFAULT.PROFILEIMAGE_HEIGHT,
            });
            const imageURL = await uploadImage(resizedImage.uri);

            setProfilePath(imageURL);
          }
          return;
        }

        if (buttonIndex === BUTTON_INDEX_LAUNCH_IMAGE_LIBLARY) {
          const image = await launchImageLibraryAsync();
          if (image && !image.cancelled) {
            const resizedImage = await resizeImage({
              imageUri: image.uri,
              maxWidth: DEFAULT.PROFILEIMAGE_WIDTH,
              maxHeight: DEFAULT.PROFILEIMAGE_HEIGHT,
            });
            const imageURL = await uploadImage(resizedImage.uri);

            setProfilePath(imageURL);
          }
        }
      },
    );
  };

  return (
    <Container>
      <StyledScrollView
        contentContainerStyle={{
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        <Wrapper>
          <TouchableOpacity
            testID="button-user-icon"
            activeOpacity={0.5}
            onPress={pressProfileImage}
          >
            {!profilePath
              ? <View style={{
                width: 90,
                height: 90,
              }}>
                <Image style={{ height: 80, width: 80 }} source={IC_PROFILE} />
                <Image style={{
                  position: 'absolute',
                  bottom: 0,
                  right: 0,
                }} source={IC_CAMERA}/>
              </View>
              : <ProfileImage
                testID="profile-image"
                source={{ uri: profilePath }}
              />
            }
          </TouchableOpacity>
          <EditText
            testID="input-nickname"
            type={EditTextInputType.DEFAULT}
            style={{ marginTop: 32 }}
            textStyle={{ color: theme.fontColor }}
            label={getString('NICKNAME')}
            placeholder={getString('NICKNAME_HINT')}
            value={nickname}
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            onChangeText={(text: string): void =>
              changeText('NICKNAME', text)
            }
          />
          <EditText
            testID="input-name"
            type={EditTextInputType.DEFAULT}
            style={{ marginTop: 32 }}
            textStyle={{ color: theme.fontColor }}
            label={getString('NAME')}
            placeholder={getString('NAME_HINT')}
            value={name}
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            onChangeText={(text: string): void =>
              changeText('NAME', text)
            }
          />
          <EditText
            type={EditTextInputType.DEFAULT}
            testID="input-status"
            style={{ marginTop: 24 }}
            textStyle={{
              color: theme.fontColor,
            }}
            label={getString('STATUS_MSG')}
            placeholder={getString('STATUS_MSG_HINT')}
            value={statusMessage}
            borderColor={theme.font}
            focusColor={theme.focused}
            placeholderTextColor={theme.placeholder}
            onChangeText={(text: string): void =>
              changeText('STATUS_MSG', text)
            }
          />
          <StyledButtonWrapper>
            <Button
              testID="button-update"
              containerStyle={{ flex: 1 }}
              style={{
                width: '100%',
                backgroundColor: theme.btnPrimary,
                borderColor: theme.btnPrimary,
                borderWidth: 1,
              }}
              textStyle={{
                color: theme.btnPrimaryFont,
                fontSize: 14,
                fontWeight: 'bold',
              }}
              // isLoading={isUpdating}
              // onPress={updateProfile}
              text={getString('UPDATE')}
            />
          </StyledButtonWrapper>
        </Wrapper>
      </StyledScrollView>
    </Container>
  );
};

export default Screen;
