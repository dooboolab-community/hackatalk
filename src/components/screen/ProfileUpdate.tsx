import React, { useEffect, useState } from 'react';
import { TouchableOpacity, View } from 'react-native';

import Button from '../shared/Button';
// import { CommonActions } from '@react-navigation/core';
import { DefaultNavigationProps } from '../../types';
import { Ionicons } from '@expo/vector-icons';
import TextInput from '../shared/TextInput';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';
import { useThemeContext } from '../../providers/ThemeProvider';

const StyledContainer = styled.View`
  flex: 1;
  background-color: ${({ theme }): string => theme.background};
  flex-direction: column;
  align-items: center;
`;

const StyledScrollView = styled.ScrollView`
  align-self: stretch;
`;

const StyledWrapper = styled.View`
  margin-top: 48px;
  width: 78%;
  flex-direction: column;
  align-items: center;
`;

const StyledBtnWrapper = styled.View`
  flex: 1;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
  align-self: stretch;
  height: 60px;
  margin-top: 20px;
  margin-bottom: 48px;
`;

interface Props {
  navigation: DefaultNavigationProps;
}

function Screen(props: Props): React.ReactElement {
  const [isUpdating, setIsUpdating] = useState(false);
  const [displayName, setDisplayName] = useState('');
  const [statusMsg, setStatusMsg] = useState('');

  useEffect(() => {
    if (isUpdating) {
      try {
        if (props.navigation) {
          props.navigation.goBack();
        }
      } catch (err) {
        // console.error(err);
      } finally {
        setIsUpdating(false);
      }
    }
  }, [isUpdating]);

  const onLogout = (): void => {
    if (props.navigation) {
      props.navigation.resetRoot({
        index: 0,
        routes: [{ name: 'AuthStack' }],
      });
    }
  };

  const onUpdate = (): void => {
    setIsUpdating(true);
  };

  const onTextChanged = (type: string, text: string): void => {
    // prettier-ignore
    switch (type) {
      case 'DISPLAY_NAME':
        setDisplayName(text);
        break;
      case 'STATUS_MSG':
        setStatusMsg(text);
        break;
    }
  };

  const onPressImg = (): void => {};

  const { theme } = useThemeContext();

  return (
    <StyledContainer>
      <StyledScrollView
        contentContainerStyle={{
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        <StyledWrapper>
          <TouchableOpacity activeOpacity={0.5} onPress={onPressImg}>
            <Ionicons
              name="ios-person"
              size={80}
              color={theme ? theme.fontColor : '#3d3d3d'}
            />
          </TouchableOpacity>
          <TextInput
            testID="input_name"
            style={{ marginTop: 80 }}
            txtLabel={getString('NAME')}
            txtHint={getString('NAME')}
            txt={displayName}
            onTextChanged={(text: string): void =>
              onTextChanged('DISPLAY_NAME', text)
            }
          />
          <TextInput
            testID="input_status"
            style={{ marginTop: 24 }}
            txtLabel={getString('STATUS_MSG')}
            txtHint={getString('STATUS_MSG')}
            txt={statusMsg}
            onTextChanged={(text: string): void =>
              onTextChanged('STATUS_MSG', text)
            }
          />
          <StyledBtnWrapper>
            <Button
              testID="logout_btn"
              onPress={onLogout}
              isWhite
              containerStyle={{ flex: 1, flexDirection: 'row' }}
            >
              {getString('LOGOUT')}
            </Button>
            <View style={{ width: 8 }} />
            <Button
              testID="update_btn"
              isLoading={isUpdating}
              onPress={onUpdate}
              containerStyle={{ flex: 1, flexDirection: 'row' }}
            >
              {getString('UPDATE')}
            </Button>
          </StyledBtnWrapper>
        </StyledWrapper>
      </StyledScrollView>
    </StyledContainer>
  );
}

export default Screen;
