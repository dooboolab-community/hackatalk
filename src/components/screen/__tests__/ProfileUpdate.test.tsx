import 'react-native';

import * as React from 'react';

import { RenderResult, render } from '@testing-library/react-native';

import ProfileUpdate from '../ProfileUpdate';
import renderer from 'react-test-renderer';

let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

const createTestProps = (obj: object) => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...obj,
});

describe('[ProfileUpdate] screen', () => {
  beforeEach(() => {
    props = createTestProps({});
    component = <ProfileUpdate {...props} />;
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    const rendered = renderer.create(component).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  it('should render [Text] with value "myText"', () => {
    const textInstance = testingLib.getByTestId('myText');
    expect(textInstance.props.children).toEqual('dooboolab');
  });

  describe('interactions', () => {
    beforeEach(() => {
      testingLib = render(component);
    });

    it('should simulate onClick', () => {
      // const btn = testingLib.queryByTestId('btn');
      // act(() => {
      //   fireEvent.press(btn);
      //   fireEvent.press(btn);
      // });
      // expect(cnt).toBe(3);
    });
  });
});
