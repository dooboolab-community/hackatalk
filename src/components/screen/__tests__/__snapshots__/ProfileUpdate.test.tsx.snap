// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test renders as expected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
    style={
      Array [
        Object {
          "alignSelf": "stretch",
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "column",
              "marginTop": 48,
              "width": "78%",
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flexDirection": "column",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Text
            focused={false}
            style={
              Array [
                Object {
                  "color": "#a3a3a3",
                  "fontSize": 12,
                  "marginBottom": 8,
                },
              ]
            }
          >
            name
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            focused={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="name"
            placeholderTextColor="rgb(175,194,219)"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderColor": "rgb(221,226,236)",
                  "borderWidth": 1,
                  "color": "black",
                  "flexDirection": "row",
                  "fontSize": 16,
                  "justifyContent": "flex-start",
                  "marginBottom": 8,
                  "paddingBottom": 22,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 22,
                },
              ]
            }
            testID="input_name"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "alignSelf": "stretch",
                "flexDirection": "column",
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <Text
            focused={false}
            style={
              Array [
                Object {
                  "color": "#a3a3a3",
                  "fontSize": 12,
                  "marginBottom": 8,
                },
              ]
            }
          >
            Status message
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            focused={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Status message"
            placeholderTextColor="rgb(175,194,219)"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "borderColor": "rgb(221,226,236)",
                  "borderWidth": 1,
                  "color": "black",
                  "flexDirection": "row",
                  "fontSize": 16,
                  "justifyContent": "flex-start",
                  "marginBottom": 8,
                  "paddingBottom": 22,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 22,
                },
              ]
            }
            testID="input_status"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginBottom": 48,
                "marginTop": 24,
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="logout_btn"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(58,139,255)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "justifyContent": "center",
                  },
                ]
              }
              white={true}
            >
              <View
                height={60}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "height": 60,
                      "justifyContent": "center",
                      "width": 136,
                    },
                  ]
                }
                width={136}
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgb(58,139,255)",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                  white={true}
                >
                  Logout
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "width": 8,
              }
            }
          />
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="update_btn"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgb(58,139,255)",
                    "borderColor": "white",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "justifyContent": "center",
                  },
                ]
              }
              white={false}
            >
              <View
                height={60}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "height": 60,
                      "justifyContent": "center",
                      "width": 136,
                    },
                  ]
                }
                width={136}
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "white",
                        "fontSize": 14,
                        "fontWeight": "bold",
                      },
                    ]
                  }
                  white={false}
                >
                  Update
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
